am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java

```diff
@@ -58,35 +65,50 @@ void whenCreatingResourceAccessMultipleEntriesAppearInDatabase() {
             JsonPointer.valueOf(""), ImmutableSet.of(CREATE, READ, UPDATE),
             JsonPointer.valueOf("/name"), ImmutableSet.of(CREATE, READ, UPDATE));
 
-        service.grantExplicitResourceAccess(createGrant(resourceId, accessorId, resourceDefinition, multipleAttributePermissions));
+        service.grantExplicitResourceAccess(createGrant(
+            resourceId, accessorId, roleName, resourceDefinition, multipleAttributePermissions));
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(2);
     }
 
     @Test
     void whenCreatingDuplicateResourceAccessEntryIsOverwritten() {
-        service.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, accessorId, resourceDefinition, ImmutableSet.of(READ)));
-        service.grantExplicitResourceAccess(createGrantForWholeDocument(resourceId, accessorId, resourceDefinition, ImmutableSet.of(READ)));
+        service.grantExplicitResourceAccess(createGrantForWholeDocument(
+            resourceId, accessorId, roleName, resourceDefinition, ImmutableSet.of(READ)));
+        service.grantExplicitResourceAccess(createGrantForWholeDocument(
+            resourceId, accessorId, roleName, resourceDefinition, ImmutableSet.of(READ)));
 
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(1);
     }
 
     @Test
     void whenCreatingResourceForMultipleUsersShouldAppearInDatabase() {
         service.grantExplicitResourceAccess(
-            createGrantForWholeDocument(resourceId, ImmutableSet.of("User1", "User2"), resourceDefinition, ImmutableSet.of(READ)));
+            createPermissionsForResourceForMultipleUsers(resourceId, ImmutableSet.of("User1", "User2")));
```

Comment:
from the story perspective, it may make sense to randomize all the test parameters even if they are sparingly used in 1 or 2 test cases (such as "user1", /"name". However, there is little value in doing it at this time but this should be taken in consideration and  used as a best practice for writing any new  integration/unit tests

@bmcts - i also see that you have already updated the tests to randomize the attributes :)